build/build_suite_test.go

```diff
@@ -362,4 +496,612 @@ var _ = Describe("Test Build utilities:", func() {
+							Name: namespace,
+						},
+					},
+					Secrets: []*v1.Secret{
+						{
+							TypeMeta: metav1.TypeMeta{
+								APIVersion: "v1",
+								Kind:       "Secret",
+							},
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "secret1",
+								Namespace: namespace,
+							},
+							Type: v1.SecretTypeBasicAuth,
+							Data: map[string][]byte{
+								v1.BasicAuthUsernameKey: []byte("username"),
+								v1.BasicAuthPasswordKey: fakePassword,
+							},
+						},
+					},
+
+					IncompleteSecrets: []*IncompleteSecret{
+						{
+							TypeMeta: metav1.TypeMeta{
+								APIVersion: "v1",
+								Kind:       "Secret",
+							},
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "incomsec1",
+								Namespace: "bar",
+							},
+							Type: "fake type",
+							Data: map[string]string{
+								"Fake Data": warningMsg,
+							},
+						},
+					},
+					System: starlingxv1.System{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "sys1",
+							Namespace: "default",
+						},
+					},
+					PlatformNetworks: []*starlingxv1.PlatformNetwork{
+						{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "pn1",
+								Namespace: "default",
+							},
+							Spec: starlingxv1.PlatformNetworkSpec{
+								Type: "mgmt",
+							},
+						},
+					},
+					AddressPools: []*starlingxv1.AddressPool{
+						{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "pn1",
+								Namespace: "default",
+							},
+							Spec: starlingxv1.AddressPoolSpec{
+								Subnet:             "192.168.204.0",
+								FloatingAddress:    &floating_address,
+								Controller0Address: &controller0_address,
+								Controller1Address: &controller1_address,
+								Prefix:             24,
+								Gateway:            &gateway,
+								Allocation: starlingxv1.AllocationInfo{
+									Order:  &allocation_order,
+									Ranges: []starlingxv1.AllocationRange{{Start: "192.168.204.2", End: "192.168.204.254"}},
+								},
+							},
+						},
+					},
+
+					DataNetworks: []*starlingxv1.DataNetwork{
+						{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "dn1",
+								Namespace: "default",
+							},
+						},
+					},
+					PtpInstances: []*starlingxv1.PtpInstance{
+						{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "ptpinst1",
+								Namespace: "default",
+							},
+						},
+					},
+					PtpInterfaces: []*starlingxv1.PtpInterface{
+						{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "ptpinf1",
+								Namespace: "default",
+							},
+						},
+					},
+					Profiles: []*starlingxv1.HostProfile{
+						{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "hostprofile1",
+								Namespace: "default",
+							},
+						},
+					},
+					Hosts: []*starlingxv1.Host{
+						{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "host",
+								Namespace: "default",
+							},
+						},
+					},
+				}
+
+				exp := `---
+apiVersion: v1
```

Comment:
wondering if this can be indented for better readability.

Reply:
It gives the yaml string with the mentioned indentation